### PR TITLE
feat: tweak pop-up state management to pass flag to Donate

### PR DESCRIPTION
**Pull request by AndyEPhipps**
#### What is it doing?
To allow us to update the tracking in Donate, we need to pass the state of the `"Help us deliver[...]"` popup; whether it's been opened _then closed_, is _currently_ open or has _never_ been shown.

#### Why is this required?
To "Add additional tracking to differentiate people using the widget on CR.COM and people coming directly to donate", to so we can "track and optimise the user journey"

#### link to Jira ticket:
[image](https://comicrelief.atlassian.net/browse/ENG-1980)

### Quick Checklist:
- [x] My PR title follows the Conventional Commit spec.

- [x] I have filled out the PR description as per the template above.

- [x] I have added tests to cover new or changed behaviour.

- [x] I have updated any relevant documentation.

### CODE REVIEW

**Review by pauldolden — APPROVED** [review]
Everything works fine here.

**Comment by AndyEPhipps** [comment]
As discussed, I need to rework this to meet the criteria:


```
When single selected and no message banner is displayed. Set dimension16 to “widget_single_na”
When monthly selected and no message banner is displayed. Set dimension16 to “widget_regular_na”
When single selected, then switched to monthly, then switched to single with the message banner shown and regular donation is made. Set dimension16 to “widget_regular_banner-shown”
When single selected, then switched to monthly, then switched to single with the message banner and banned closed, then single donation is made. Set dimension16 to “widget_regular_banner-closed”
```

File: src/utils/Membership.js [hunk 1]
```diff
@@ -56,7 +58,7 @@ const handleDonateSubmission = (
   }
   const givingTypeUrl = givingType === 'single' ? 'single' : 'monthly';
 
-  window.location.href = `${donateLink}?clientOverride=${clientID}&amount=${amount}&currency=GBP&givingType=${givingTypeUrl}&cartId=${cartID}&affiliate=${affiliateValue}&siteurl=${currentpageUrl}&rowID=${mbshipID}`;
+  window.location.href = `${donateLink}?clientOverride=${clientID}&amount=${amount}&currency=GBP&givingType=${givingTypeUrl}&cartId=${cartID}&affiliate=${affiliateValue}&siteurl=${currentpageUrl}&rowID=${mbshipID}&popUpOpen=${popOpen}&&popUpShown=${popUpShown}`;
```
Comment (pauldolden) [inline src/utils/Membership.js:61]: Is this supposed to be `&` rather than `&&`?

Comment (AndyEPhipps) [reply src/utils/Membership.js:61]: lol whoops

can't get enough of those boolean operators

**Comment by AndyEPhipps** [comment]
Caught up with Hiral, and he realised that we really don't need to be passing the "is the popup CURRENTLY open" flag, so have paired it back to just use the "has the popup EVER been shown" flag.

**Pull request merged**